### DISCUSSION TITLE
Is it possible to use Nyx to publish in remote Bitbucket?

### DISCUSSION THREAD

**Discussion by eborreo** [mooltiverse/nyx#192] [Q&A]
I wonder if Nyx tool is avaible for service type bitbucket? or only for GitHub? Thank you

**Comment by flelli** [comment]
You can push Git changes to Bitbucket repos but non-git features are not suported (yet).
What is your use case?

**Reply by eborreo** [reply]
thank you so much for your quick response. 
I came across your documentation about  Semantic Release Automation with Gradle using Nyx https://levelup.gitconnected.com/semantic-release-automation-with-gradle-using-nyx-ba345235a365 
and this part is a question for me being a bitbucket repo user,  

```
  services {
    github {
      type = 'GITHUB'
      options {
        AUTHENTICATION_TOKEN = '{{#environmentVariable}}GH_TOKEN{{/environmentVariable}}'
        REPOSITORY_NAME = 'myrepo'
        REPOSITORY_OWNER = 'jdoe'
      }
    }
```

looks like service types are only  available for Github and Gitlab based also from what Ive read here https://mooltiverse.github.io/nyx/guide/user/configuration-reference/services/ ?

**Reply by flelli** [reply]
Yes, those "extra" services (like release publication) are only available for GitHub and GitLab so far.
Which ones would you need for Bitbucket?

**Reply by eborreo** [reply]
how is git commit or push possible if publicationservice is github and not bitbucket?

`releaseTypes {
    publicationServices = [ 'github' ]
    items {
      mainline {
        ...
        description = '{{#fileContent}}build/CHANGELOG.md{{/fileContent}}'
        gitCommit = 'true'
      }
    }
  }`

**Reply by eborreo** [reply]
Ah I see, when you say release publication it is actually the package available for other people to download.
I thought that service is use to connect to remote repo so that you can push your local commit..

**Reply by flelli** [reply]
yes, release publication is about publishing pages like [this one](https://github.com/mooltiverse/nyx/releases), and this is not a Git feature.
For all about committing, tagging and pushing you don't need that and you can do that also with Bitbucket using the standard Git repo features

**Reply by eborreo** [reply]
when you say 'using standard Git repo features' - is it the manual way? or you mean by using the release type attributes as listed here
https://mooltiverse.github.io/nyx/guide/user/configuration-reference/release-types/
and to connect to Git should follow this document - https://mooltiverse.github.io/nyx/guide/user/configuration-reference/git/ ?

or the git Attribute (like gitPush, gitcommit, gitTag) of release type is only applicable to release publication? 

is it correct that "releaseTypes/publicationServices" is for release publication and "releaseTypes/remoteRepositories" is when you want to run standard git repo features ?
if so for remoteRepositories, we can have git config as:
```
  git {
    remotes {
      origin {
        authenticationMethod = 'USER_PASSWORD'
        user = 'jdoe'
        password = 'somepassword'
      }
```
in under releaseTypes must have this:
```
  releaseTypes {
    enabled = [ 'mainline', 'maturity', 'integration', 'hotfix', 'feature', 'release', 'maintenance', 'internal' ]
    publicationServices = [ ]
    remoteRepositories = [ 'origin', 'replica' ]
```
publicationServices is empty since we are doing any github release
is that right?

**Reply by flelli** [reply]
confirmed, all of the above is correct

**Reply by eborreo** [reply]
Thanks Flelli :)
One last questionm Is this also applicable to work in CI/CD azure devops? Thanks

**Comment by flelli** [comment]
I never tried that but as long as you use the Gradle plugin or the command line I can't see any issues. Let me know how it goes!
